pyeudiw/oauth2/dpop.py

```diff
@@ -81,12 +80,24 @@ def __init__(
             if self.dpop_header_prefix in http_header_authz
             else http_header_authz
         )
+        # If the jwt is invalid, this will raise an exception
+        try:
+            unpad_jwt_header(http_header_dpop)
+        except UnicodeDecodeError:
+            raise ValueError("DPoP proof is not a valid JWT")
         self.proof = http_header_dpop
 
     @property
     def is_valid(self):
         jws_verifier = JWSHelper(self.public_jwk)
-        dpop_valid = jws_verifier.verify(self.proof)
+        try:
+            dpop_valid = jws_verifier.verify(self.proof)
+        except Exception as e:
```

Comment:
Please specialize this exception,

1. define a new exception in exceptions.py
2. use that exception here

ps = I like the generalization `{e.__class__.__name__}`

Reply:
Resolved in 5b531ed